File: src/com/Torvald/Terrarum/ConsoleCommand/Echo.java
package com.Torvald.Terrarum.ConsoleCommand;

import com.Torvald.Terrarum.Terrarum;
import com.Torvald.Terrarum.UserInterface.ConsoleWindow;

import java.util.Arrays;
import java.util.List;

/**
 * Created by minjaesong on 16-01-16.
 */
class Echo implements ConsoleCommand {
    @Override
    public void execute(String[] args) {
        String[] argsWoHeader = new String[args.length - 1];
        for (int i = 0; i < argsWoHeader.length; i++)
            argsWoHeader[i] = args[i + 1];

        Arrays.asList(argsWoHeader).forEach(
                ((ConsoleWindow) Terrarum.game.consoleHandler.getUI())::sendMessage
        );
    }

    public void execute(String single_line) {
        ((ConsoleWindow) Terrarum.game.consoleHandler.getUI())
                .sendMessage(single_line);
    }

    @Override
    public void printUsage() {

    }
}

File: src/net/torvald/terrarum/console/Echo.kt
package net.torvald.terrarum.console

import net.torvald.terrarum.*
import net.torvald.terrarum.App.csi0
import net.torvald.terrarum.App.csiG
import net.torvald.terrarum.ui.ConsoleWindow

/**
 * Created by minjaesong on 2016-01-16.
 */
internal object Echo : ConsoleCommand {
    override fun execute(args: Array<String>) {
        val argsWoHeader = Array<String>(args.size - 1, {it -> args[it + 1]})
        argsWoHeader.forEach { execute(it) }
    }

    fun execute(single_line: String) {
        val sb = StringBuilder()
        for (ch in single_line) {
            if (ch == '\n') {
                INGAME.consoleHandler.sendMessage(sb.toString())
                println("[Echo] $csiG$sb$csi0")
                sb.delete(0, sb.length - 1)
            }
            else
                sb.append(ch)
        }
        INGAME.consoleHandler.sendMessage(sb.toString())
        println("[Echo] $csiG$sb$csi0")
    }

    operator fun invoke(args: Array<String>) = execute(args)
    operator fun invoke(single_line: String) = execute(single_line)

    override fun printUsage() {

    }
}
